src/pages/WithdrawAndClose.vue

```diff
@@ -0,0 +1,189 @@
+import { closeProposal, signTxUsingLsig } from "@/utility";
+import moment from "moment";
+
+export default defineComponent({
+	name: "WithdrawAndClose",
+	data() {
+		return {
+			error: "",
+			key: "WithdrawAndCloseKey",
+			CLOSE_PROPOSAL_DESCRIPTION,
+			WITHDRAW_FROM_PROPOSAL_DESCRIPTION,
+		};
+	},
+
+	setup() {
+		const daoIDStore = reactive(DaoID());
+		const walletStore = reactive(WalletStore());
+		const proposalStore = reactive(ProposalStore());
+		const currentTime = moment(new Date()).unix();
+		return {
+			daoIDStore,
+			walletStore,
+			validateMessages: VALIDATE_MESSAGES,
+			proposalStore,
+			currentTime,
+		};
+	},
+	methods: {
+		errorDescription() {
+			if (this.proposalStore.voting_end > this.currentTime) {
+				return "Voting period has not yet finished.";
+			}
+		},
+		checkWithdrawAndCloseValidity() {
+			// it’s past execution: proposal.executed == 1  || proposal.execute_before < now
+			if (
+				this.proposalStore.executed === 1 ||
+				this.proposalStore.execute_before < this.currentTime
+			) {
+				return true;
+			}
+			return false;
+		},
+		async withdrawTokens() {
+			const proposalLsig: LogicSigAccount = await getProposalLsig(
+				this.daoIDStore.dao_id as number,
+				this.walletStore.address
+			);
+			loadingMessage(this.key);
+			const minimumDepositToken = this.daoIDStore.global_app_state?.get(
+				GLOBAL_STATE_MAP_KEY.Deposit
+			);
+			console.log(
+				`* Withrawing from proposalLsig ${minimumDepositToken} amount by ${this.walletStore.address} *`
+			);
+			// can only be withdrawn if the curent user is the proposer
+			if (proposalLsig.address() === this.proposalStore.proposal_addr) {
+				try {
+					const withdrawParam: types.ExecParams = {
+						type: types.TransactionType.TransferAsset,
+						sign: types.SignType.LogicSignature,
+						fromAccountAddr: proposalLsig.address(),
+						toAccountAddr: this.walletStore.address,
+						amount: minimumDepositToken as number,
+						lsig: proposalLsig,
+						assetID: this.daoIDStore.govt_id as number,
+						payFlags: { totalFee: 1000 },
+					};
+					try {
+						const response = await signTxUsingLsig(proposalLsig, withdrawParam);
+						console.log("withdraw from proposal", response);
+						searchApplicationAndAccount(); // to update locked and available token on UI
+						successMessage(this.key);
+						openSuccessNotificationWithIcon(
+							SUCCESSFUL,
+							withdrawTokenMessage.SUCCESSFUL(minimumDepositToken as number)
+						);
+					} catch (error) {
+						errorMessage(this.key);
+						this.error = error.message;
+						console.error("Transaction Failed", error);
+					}
+				} catch (error) {
+					openErrorNotificationWithIcon(UNSUCCESSFUL, error.message);
+				}
+			} else {
+				openErrorNotificationWithIcon(
+					UNSUCCESSFUL,
+					"Only creator of the proposal can withdraw from the proposal."
+				);
+			}
+		},
+		async handleCloseProposal() {
+			const proposalLsig: LogicSigAccount = await getProposalLsig(
```

Comment:
Let's do one thing. We are using this getProposalLsig almost every place. Let's directly use it inside the method instead of here. From handleCloseProposal we will pass the required param.

Reply:
I don't think that would be the right approach, because we need it here for checking current address (to know if it is proposer address), and that is something specific to UI, so it should be here instead of common functions.
What do you think..?